src/portal/error.rs

```diff
@@ -116,3 +131,25 @@ impl std::fmt::Display for InvalidItemError {
         )
     }
 }
+#[derive(Debug, Clone)]
+pub enum WeakKeyError {
```

Comment:
Could use a bit of documentation, at least the type itself

Reply:
Added the reasons why we do those checks as well.